packages/cubejs-server-core/src/core/server.ts

```diff
@@ -204,7 +204,9 @@ export class CubejsServerCore {
     this.compilerCache = new LRUCache<string, CompilerApi>({
       max: this.options.compilerCacheSize || 250,
       ttl: this.options.maxCompilerCacheKeepAlive,
-      updateAgeOnGet: this.options.updateCompilerCacheKeepAlive
+      updateAgeOnGet: this.options.updateCompilerCacheKeepAlive,
+      // needed to clear the setInterval timer for proactive cache internal cleanups
+      dispose: (v) => v.dispose(),
```

Comment:
👍 

Then we need to reset `compilerCache` to trigger `dispose` on each `CompilerApi.` 

Right now, we don't clean it during the `shutdown`. As the last step, we should call `this.compilerCache.reset()` in the `shutdown` method.

Reply:
yep, sure! will do!